src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+            final int mask = 0x1 << shift;
+            out = (byte) ((out & ~mask) | bits);
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts a long into an array of int using the default (little endian, Lsb0) byte and bit
+     * ordering.
+     * </p>
+     *
+     * @param src the long to convert
+     * @param srcPos the position in {@code src}, in bits, from where to start the conversion
+     * @param dst the destination array
+     * @param dstPos the position in {@code dst} where to copy the result
+     * @param nInts the number of ints to copy to {@code dst}, must be smaller or equal to the
+     *            width of the input (from srcPos to msb)
+     * @return {@code dst}
+     * @throws NullPointerException if {@code dst} is {@code null} and {@code nInts > 0}
+     * @throws IllegalArgumentException if {@code (nInts-1)*32+srcPos >= 64}
+     * @throws ArrayIndexOutOfBoundsException if {@code dstPos + nInts > dst.length}
+     */
+    public static int[] longToIntArray(final long src, final int srcPos, final int[] dst, final int dstPos,
+            final int nInts) {
+        if (0 == nInts) {
+            return dst;
+        }
+        if ((nInts - 1) * 32 + srcPos >= 64) {
+            throw new IllegalArgumentException("(nInts-1)*32+srcPos is greater or equal to than 64");
+        }
+        for (int i = 0; i < nInts; i++) {
+            final int shift = i * 32 + srcPos;
+            dst[dstPos + i] = (int) (0xffffffff & (src >> shift));
+        }
+        return dst;
+    }
+
+    /**
+     * <p>
+     * Converts a long into an array of short using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the long to convert
+     * @param srcPos the position in {@code src}, in bits, from where to start the conversion
+     * @param dst the destination array
+     * @param dstPos the position in {@code dst} where to copy the result
+     * @param nShorts the number of shorts to copy to {@code dst}, must be smaller or equal to
+     *            the width of the input (from srcPos to msb)
+     * @return {@code dst}
+     * @throws NullPointerException if {@code dst} is {@code null}
+     * @throws IllegalArgumentException if {@code (nShorts-1)*16+srcPos >= 64}
+     * @throws ArrayIndexOutOfBoundsException if {@code dstPos + nShorts > dst.length}
+     */
+    public static short[] longToShortArray(final long src, final int srcPos, final short[] dst, final int dstPos,
+            final int nShorts) {
+        if (0 == nShorts) {
+            return dst;
+        }
+        if ((nShorts - 1) * 16 + srcPos >= 64) {
+            throw new IllegalArgumentException("(nShorts-1)*16+srcPos is greater or equal to than 64");
+        }
+        for (int i = 0; i < nShorts; i++) {
+            final int shift = i * 16 + srcPos;
+            dst[dstPos + i] = (short) (0xffff & (src >> shift));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).